js/foam/u2/UpdateView.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+CLASS({
+  package: 'foam.u2',
+  name: 'UpdateView',
+  extends: 'foam.u2.View',
```

Comment:
Having the functionality implemented to attach to the model, I agree, but
also (perhaps in a separate package) it's convenient to have a view that
just does the right thing so you can define a UI and saving behaviour
without having to fiddle with your data.

We're entering EasyDAO (EasyView?) territory. It would certanily make
UpdateDetailView much smaller :P

On Tue, Nov 24, 2015 at 8:47 AM adamvy <notifications@github.com> wrote:

> In js/foam/u2/UpdateView.js
> <https://github.com/foam-framework/foam/pull/473#discussion_r45736505>:
>
> > + * Licensed under the Apache License, Version 2.0 (the "License");
> > + * you may not use this file except in compliance with the License.
> > + * You may obtain a copy of the License at
> > + *
> > + *     http://www.apache.org/licenses/LICENSE-2.0
>
>
> > + *
> > + * Unless required by applicable law or agreed to in writing, software
> > + * distributed under the License is distributed on an "AS IS" BASIS,
> > + * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > + * See the License for the specific language governing permissions and
> > + * limitations under the License.
> > + */
> > +CLASS({
> > +  package: 'foam.u2',
> > +  name: 'UpdateView',
> > +  extends: 'foam.u2.View',
>
> The question of why does it need to be a view is broader. Why does it need
> to live in the View tree at all?
>
> You 're decorating the view to apply some sort of model saving logic. But
> why not just decorate the model (MVC) instead? This is a model concern in
> MVC, not a view concern.
>
> Using something like the foam.persistence.ObjectReplicator you apply it to
> your model itself, and it gives you a new model which will auto-save to a
> dao when changed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/foam-framework/foam/pull/473/files#r45736505>.
>